web3.js/test/url.ts

```diff
@@ -11,5 +19,16 @@ export const wsUrl = process.env.TEST_LIVE
   ? 'ws://localhost:8900/'
   : 'ws://localhost:9999/';
 
+const nodeVersion = Number(process.version.split('.')[0]);
+
+export const Controller = function () {
+  if (nodeVersion < 16) {
+    import('node-abort-controller').then(importedModule => {
```

Comment:
Making this async is begging for trouble. Since this is a test, feel free to just import both implementations, and choose the right one.

```ts
import { AbortController as AbortControllerPolyfill } from 'node-abort-controller';

const AbortController = nodeVersion >= 16
  ? AbortController
  : AbortControllerPolyfill;
```

I still wonder; is this broken, in general, for Node 14? If so, we should probably just drop support for Node 14.

Reply:
Good point, it seems as though "node-abort-contoller" can't be defined in the same file as AbortController (node 16) from Typescripts perspective -- not entirely sure why, but I believe the types differ between the AbortControllers causing some type of overflow error.

Its a little hacky, but we can get around it by defining something like this in any file that isn't the controller test:
`export const Node14Controller = function() {
  return new AbortControllerPolyfill();
}`

Then in the controller test:
`const abortController: any = (nodeVersion >= 16) ? new AbortController() : Node14Controller();`

Alternative would be if node14 isn't needed, the normal AbortController in node16 works fine with Typescript/the tests and we would just drop node-abort-contoller.

Reply:
> …it seems as though "node-abort-contoller" can't be defined in the same file as AbortController (node 16) from Typescripts perspective…

Oof. Just cast it.

```ts
const AbortController = (nodeVersion >= 16
  ? AbortController
  : AbortControllerPolyfill) as unknown as typeof AbortController;
```

…or whatever.